File: matlab/fxdpt_2.m
function fxdpt_2(x_0, n)
% This function performs newton's method.
% x_0 => initial value
% n   => iterations

% First we compute the formulas for f(x) and f'(x).
% The '@' symbol tells MATLAB to plug in x for the function.
f = @(x) x^3 + 4*x^2 - 10;
fp = @(x) 3*x^2 + 8*x;

% Next, we form the iteration function g(x).
g = @(x) x - (f(x)/fp(x));    % Newton's method
g_1 = @(x) x - f(x);          
g_2 = @(x) sqrt(10/x - 4*x);
g_3 = @(x) 1/2 * sqrt(10-x^3);
g_4 = @(x) sqrt(10/(4+x))

% The following "for" loop performs the iteration.
% The vector "r" records the values of the iteration.
% The first value of "r" will be the user input x_0.
r(1) = x_0;
for k=2:n
    r(k) = g( r( k-1 ) );
end

% The results are reported as a column vector r'.
Iteration_Vector = r'

end

File: python/fxdpt_2.py
# -*- coding: utf-8 -*-
"""
Created on Wed Feb 19 08:37:09 2020

@author: cpjohnson

This program implements a simple version of Newton's Method
  The input required is x_0 and number of iterations n
"""
import numpy as np

def f(x):
    return x**3 + 4*x**2 - 10

def fp(x):
    return 3*x**2 + 8*x

def g_1(x):
    return x - f(x)/fp(x) # Newton's method

def g_2(x):
    return np.sqrt(10.0/x - 4*x)

def g_3(x):
    return 0.5 * np.sqrt(10 - x**3)

def g_4(x):
    return np.sqrt(10/(4+x))

def fxdpt_2(x_0, n):
    # x_0 is the initial guess
    # n is the number of iterations
    print(x_0)
    for k in range(1, n):
        x_0 = g_3(x_0)
        print("iteration {}: {}".format(k, x_0))
        
